### PR TITLE
Drop x/crypto package dependency

### LINKED ISSUE #928: Dropping dependency on golang.org/x/crypto
This library is pulled for only one check - to see if output is a terminal or not.

Would it be possible just to inline one used function from it? This would allow us (and all people who pulls it in) to drop the dependency on it as well. The import with "crypto" in it triggers additional attention from our legal department, and we have to explain that it does not have anything "cryptographic" in it - just a check for is it a terminal or not.

Ideally, `ssh/terminal` should be not under crypto , but I will try to ping golang folks about it directly.

### PR DESCRIPTION
**Pull request by tandr**
Closes #928

### CODE REVIEW

File: internal/terminal/terminal_check_bsd.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+package terminal
+
+import "golang.org/x/sys/unix"
+
+const ioctlReadTermios = unix.TIOCGETA
+
+func IsTerminal(fd int) bool {
+	_, err := unix.IoctlGetTermios(fd, ioctlReadTermios)
+	return err == nil
+}
```
Comment (lavoiesl) [inline internal/terminal/terminal_check_bsd.go]: Newline

File: internal/terminal/terminal_check_unix.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+package terminal
+
+import "golang.org/x/sys/unix"
+
+const ioctlReadTermios = unix.TCGETS
+
+func IsTerminal(fd int) bool {
+	_, err := unix.IoctlGetTermios(fd, ioctlReadTermios)
+	return err == nil
+}
```
Comment (lavoiesl) [inline internal/terminal/terminal_check_unix.go]: Newline

Comment (tandr) [reply internal/terminal/terminal_check_unix.go]: added

File: internal/terminal/terminal_check_bsd.go (hunk 1 above)
Comment (tandr) [reply internal/terminal/terminal_check_bsd.go]: added

**Comment by tandr** [comment]
Would it be possible to slap a tag on it once it is merged in? Or it is not sufficiently big change to bump  it to v1.4.1 ?

**Comment by lavoiesl** [comment]
A patch bump would be in order, yes. I’m just waiting for someone else to review.

**Comment by dgsb** [comment]
@tandr that looks good to me.

**Comment by tandr** [comment]
@dgsb David, would you be able to tell when do you guys expect v1.4.1 get tagged? Reason I am asking is that or I need to pull it now and then second time when you tag it, or if you expect it tomorrow or in very near future, I can just wait for it instead.

**Comment by thaJeztah** [comment]
+1 on tagging a v1.4.1 for this 🤗

**Comment by dgsb** [comment]
@tandr and @thaJeztah we can not yet, we do not want to make public the new terminal interface.

**Pull request merged**